browser-tests/header/header.components.ts

```diff
@@ -38,25 +38,15 @@ export const findHeader = async (
   };
 
   const languageSelector = () => {
-    const selectors = {
-      languageSelector() {
-        return withinHeader().findByRole('button', {
-          name: getTranslations(currentLang).navigation
-            .languageSelectorAriaLabel,
-        });
-      },
-      languageSelectorItem(lang: SUPPORTED_LANGUAGES) {
-        return withinHeader().findByRole('link', {
-          name: getTranslations(currentLang).navigation.languages[lang],
-        });
-      },
-    };
-
     const actions = {
       async changeLanguage(lang: SUPPORTED_LANGUAGES) {
-        const result = await t
-          .click(selectors.languageSelector())
-          .click(selectors.languageSelectorItem(lang));
+        const result = await t.click(
+          withinHeader().findByRole('button', {
+            hidden: false,
+            name: /in english/i,
+          })
+        );
```

Comment:
I think that this change makes `changeLanguage` function name a little bit misleading since it's always changing language to English. 
You could return the languageSelectorItem selector and use it to get desired language button. In that case you have to change the query role from link to button e.g. 
```
const languageSelector = () => {
    const selectors = {
      languageSelectorButton(lang: SUPPORTED_LANGUAGES) {
        return withinHeader().findByRole('button', {
          name: getTranslations(currentLang).navigation.languages[lang],
        });
      },
    };

    const actions = {
      async changeLanguage(lang: SUPPORTED_LANGUAGES) {
        const result = await t.click(selectors.languageSelectorButton(lang));

        currentLang = lang;
        setDataToPrintOnFailure(t, 'expectedLanguage', lang);
        return result;
      },
    };

    return { actions };
  };
```

Reply:
https://github.com/City-of-Helsinki/linkedcomponents-ui/pull/262/commits/b34eaeb16bc3ca3fae394473bdde05620a1c2ae3